service/src/test/java/tds/config/services/impl/ConfigServiceImplTest.java

```diff
@@ -0,0 +1,275 @@
+
+        assertThat(windows).containsExactly(window2);
+    }
+
+    @Test
+    public void shouldReturnDistinctFormWindowsByWindowIdAndFormKey() {
+        AssessmentWindow window = new AssessmentWindow.Builder()
+            .withWindowId("id")
+            .withFormKey("formKey")
+            .withAssessmentId("SBAC-Mathematics-8")
+            .build();
+
+        AssessmentWindow window2 = new AssessmentWindow.Builder()
+            .withWindowId("id2")
+            .withFormKey("formKey")
+            .withAssessmentId("SBAC-Mathematics-8-2018")
+            .build();
+
+        AssessmentWindowParameters properties = new AssessmentWindowParameters.Builder(23, "SBAC_PT", "SBAC-Mathematics-8", 0)
+            .withFormList("id:formKey")
+            .build();
+        AssessmentFormWindowProperties assessmentFormWindowProperties = new AssessmentFormWindowProperties(true, true, "formField", true);
+
+        when(mockAssessmentWindowQueryRepository.findCurrentAssessmentFormWindows("SBAC_PT", "SBAC-Mathematics-8", 0, 0, 0, 0, 0)).thenReturn(Arrays.asList(window, window2));
+        when(mockAssessmentWindowQueryRepository.findAssessmentFormWindowProperties("SBAC_PT", "SBAC-Mathematics-8", 0)).thenReturn(Optional.of(assessmentFormWindowProperties));
+        List<AssessmentWindow> windows = configService.findAssessmentWindows(properties);
+
+        assertThat(windows).containsExactly(window);
+    }
+
+    private Optional<ClientTestProperty> getMockClientTestProperty() {
+        ClientTestProperty clientTestProperty = new ClientTestProperty.Builder()
+                .withClientName("SBAC_PT")
+                .withAssessmentId("TEST_ID")
+                .withMaxOpportunities(3)
+                .withPrefetch(2)
+                .withIsSelectable(true)
+                .withLabel("label")
+                .withSubjectName("subject name")
+                .withInitialAbilityBySubject(true)
+                .withAccommodationFamily("accommodation family")
+                .withSortOrder(1)
+                .withRtsFormField("rts form field")
+                .withRequireRtsWindow(true)
+                .withRequireRtsMode(true)
+                .withRequireRtsModeWindow(true)
+                .withDeleteUnansweredItems(true)
+                .withAbilitySlope(5.0D)
+                .withAbilityIntercept(10.0D)
+                .withValidateCompleteness(true)
+                .withGradeText("grade text")
+                .build();
+
+        return Optional.of(clientTestProperty);
+    }
+
+    private List<ClientSystemFlag> getMockClientSystemFlag() {
+        List<ClientSystemFlag> clientSystemFlags = new ArrayList<>();
+
+        for (Integer i = 0; i < 5; i++) {
+            ClientSystemFlag clientSystemFlag = new ClientSystemFlag.Builder()
+                    .withClientName("SBAC_PT")
+                    .withAuditObject("AUDIT_OBJECT " + i.toString())
```

Comment:
I'm not 100% sure why this is green as this isn't code I wrote.  That being said, the Object Integer is unnecessary.  I'll change to primitive.